integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMonitoringExporter.java

```diff
@@ -119,24 +130,88 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
    * Verify server pods are in ready state and services are created.
    * Verify login to WebLogic console is successful thru istio ingress port.
    * Deploy a web application thru istio http ingress port using REST api.
-   * Access web application thru istio http ingress port using curl.
    * Deploy Istio provided Promethues
    * Verify Weblogic metrics can be processed via istio based prometheus
    */
   @Test
   @DisplayName("Create istio provided prometheus and verify "
       + "it can monitor Weblogic domain via weblogic exporter webapp")
-  public void testIstioPrometheus() {
-    assertDoesNotThrow(() -> setupIstioModelInImageDomain(), "setup for istio based domain failed");
-    int prometheusPort = getNextFreePort();
-    assertTrue(deployIstioPrometheus(domainNamespace, domainUid,
-        String.valueOf(prometheusPort)), "failed to install istio prometheus");
+  public void testIstioPrometheusViaExporterWebApp() {
+    assertDoesNotThrow(() -> downloadMonitoringExporterApp(RESOURCE_DIR
+        + "/exporter/exporter-config.yaml", RESULTS_ROOT), "Failed to download monitoring exporter application");
+    String miiImage = createAndVerifyMiiImageWithMonitoringExporter(RESULTS_ROOT + "/wls-exporter.war",
+        MODEL_DIR + "/model.monexp.yaml");
+    String managedServerPrefix = domain1Uid + "-cluster-1-managed-server";
+    assertDoesNotThrow(() -> setupIstioModelInImageDomain(miiImage,
+        domain1Namespace,domain1Uid, managedServerPrefix), "setup for istio based domain failed");
+    if (!isPrometheusDeployed) {
+      assertTrue(deployIstioPrometheus(domain1Namespace, domain1Uid,
+          String.valueOf(prometheusPort)), "failed to install istio prometheus");
+      isPrometheusDeployed = true;
+    } else {
+      String oldRegex = String.format("regex: %s;%s", domain2Namespace, domain2Uid);
+      String newRegex = String.format("regex: %s;%s", domain1Namespace, domain1Uid);
+      assertDoesNotThrow(() -> editPrometheusCM(oldRegex, newRegex, "istio-system","prometheus"),
+          "Can't modify Prometheus CM, not possible to monitor " + domain1Uid);
+    }
 
     //verify metrics via prometheus
-    String testappPrometheusSearchKey =
-        "wls_servlet_invocation_total_count%7Bapp%3D%22testwebapp%22%7D%5B15s%5D";
-    assertDoesNotThrow(() -> checkMetricsViaPrometheus(testappPrometheusSearchKey,
-        "testwebapp", prometheusPort));
+    String sessionAppPrometheusSearchKey =
+        "wls_servlet_invocation_total_count%7Bapp%3D%22myear%22%7D%5B15s%5D";
+    assertDoesNotThrow(() -> checkMetricsViaPrometheus(sessionAppPrometheusSearchKey, "sessmigr",prometheusPort));
```

Comment:
Add a assert failure message in case if the assert fails

Reply:
done